src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -165,80 +177,65 @@ export default class AmalgamatingBusinesses extends Mixins(CommonMixin) {
-    })
+      LegalServices.fetchBusinessInfo(businessLookup.identifier),
+      LegalServices.fetchAddresses(businessLookup.identifier),
+      LegalServices.fetchFilings(businessLookup.identifier)
+    ]).then(results => results.map((result: any) => result.value))
 
-    if (data.length === 3) {
-      business = data[0].data?.business
-      business.businessContact = data[1].contacts[0]
-      business.officeAddress = data[2]
+    // check if all elements are undefined
+    if (!data.some(d => d)) {
+      this.snackbarText = 'Unable to add that business.'
+      this.snackbar = true
+      return
     }
 
-    // If the business is not null (LEAR Entity), create from it a TING business following the interface.
-    // If the amalgamating businesses array is not empty, check if identifier already exists.
-    // If identifier already exists, don't add the business to the array.
-    if (business) {
-      const amalgamatingBusinesses = this.getAmalgamatingBusinesses
+    const authInfo = data[0]
+    const businessInfo = data[1]
+    const addresses = data[2]
+    const filings = data[3]
 
-      const tingBusiness = {
-        type: 'lear',
-        role: AmlRoles.AMALGAMATING,
-        identifier: business.identifier,
-        name: business.legalName,
-        email: business.businessContact.email,
-        legalType: business.legalType,
-        address: business.officeAddress.registeredOffice.mailingAddress,
-        goodStanding: business.goodStanding
-      } as AmalgamatingBusinessIF
-
-      if (!amalgamatingBusinesses.find((b: any) => b.identifier === business.identifier)) {
-        amalgamatingBusinesses.push(tingBusiness)
+    // If identifier already exists, don't add the business to the array.
+    if (this.getAmalgamatingBusinesses.find((b: any) => b.identifier === businessInfo.identifier)) {
+      this.snackbarText = 'Business is already in table.'
+      this.snackbar = true
+      return
+    }
 
-        // Set the new amalgamated businesses array in the store.
-        this.setAmalgamatingBusinesses(amalgamatingBusinesses)
-      }
+    // Create amalgamating business object.
+    // If we couldn't fetch some data, use business lookup data instead.
+    const tingBusiness: AmalgamatingBusinessIF = {
+      type: AmlTypes.LEAR,
+      role: AmlRoles.AMALGAMATING,
+      identifier: businessInfo?.identifier || businessLookup.identifier,
+      name: businessInfo?.legalName || businessLookup.name,
+      email: authInfo?.contacts[0]?.email, // may be undefined
+      legalType: businessInfo?.legalType || businessLookup.legalType,
+      address: addresses?.registeredOffice?.mailingAddress, // may be undefined
+      isNotInGoodStanding: (businessInfo?.goodStanding === false),
+      isFutureEffective: (filings[0]?.isFutureEffective === true),
+      isLimitedRestoration: (filings[0]?.filingSubType === RestorationTypes.LIMITED) ||
+        (filings[0]?.filingSubType === RestorationTypes.LTD_EXTEND)
     }
```

Comment:
The above works when we started from a business lookup, but I don't think it will work when resuming a draft.

I wrote the above to provide a fallback when "some" fetches failed (ie, to handle "not affiliated"). I think we need to review this and tighten it up -- only allow a specific "not affiliated" case through so we don't have to fall back for everything. This should allow a simple re-fetch when resuming a draft with no need to fall back.

^^ Maybe a future PR?

Reply:
I think it should work even when resuming a draft. I tried it and it seems fine!

Makes sense, future PR sounds good.

Reply:
But when resuming a draft, there won't be a business lookup object to fall back to.

I'm thinking this will work out in the wash when we tighten up what fallbacks we allow (ie, only for not affiliated if possible).

Reply:
We may still want to refactor what I wrote so you can reuse it when resuming a draft, but I think we can work with this:
- if there's no auth info and user is staff then don't add the business to the table
- if there's no auth info and user is not staff then add the business
- if any of the Legal API calls fails then don't add the business to the table

I would output a console log if a business in the draft can't be added.